### ISSUE TITLE
FieldArrayType add() function creates issues with custom subscriptions to valueChanges

### ISSUE THREAD

**Issue by iMantasas** [ngx-formly/ngx-formly#2061]
**Description**
I have some need to have subscriptions on my form controls valueChanges observable. At first, if the form is created and I subscribe to valueChanges in the onInit, everything works as expected, and the form's model contains correct values. If I add a element, the same subscription reaches the old model value, because the model's updating subscription is run AFTER my custom subscription.

Can something really be done about this, or not really? I've got a hack written up for now, but I don't really like it, and would prefer it working out of the box somehow.

**Minimal Reproduction**
https://stackblitz.com/edit/angular-4zesha

**Your Environment**
- Angular version: 8.2.14
- Formly version: 5.5.8

**Additional context**

**Comment by aitboudad** [comment]
Hi,
I didn't figure out what's going wrong based on the provided example, can you simplify it (real use case would be better) in order to illustrate what you've described in the description.
Thanks in advance!

**Comment by iMantasas** [comment]
Hi,

Thanks for the quick response. Looks like I didn't actually save the last change I did, so the code in the link is updated now, and easier to see. To reproduce issue:

1. Type something in Testing field, the value is updated right away.
2. Add a new investment (to trigger add())
3. Type something in Testing field, the value will be updated with 1 value delay.

In the console log you can see that the subscription index in the observers goes from 1 to 0 as soon as you add a new row in repeat (makes sense, creates the issue).

**Comment by aitboudad** [comment]
I see it now, the order of subscribing changes when adding a new field.

**Comment by iMantasas** [comment]
The only way I see of fixing this is NOT rebuilding all form and recreating all subscriptions when a FieldArray new row gets added, but I wasn't sure whether this might/might not have any side effects.

**Comment by aitboudad** [comment]
Well, I think this issue is already fixed in master but it may take some time to be released (`v6`), for a fix forcing `formly` subscribing order somehow would be a better solution for now.

**Comment by iMantasas** [comment]
Well, I guess I'll wait for v6 and see then. Thanks!

For anyone who might run into the same issue, what I've done for now (really not nice, but does the trick) is:

```
createSubscription() {
    this.formControl.valueChanges
      .pipe(
        // HACK to ensure that this subscription gets evaluated LAST
        tap((val) => {
          // @ts-ignore
          if(this.formControl.valueChanges.observers.indexOf(this.subscription) !== this.formControl.valueChanges.observers.length - 1) {
            this.subscription.unsubscribe();
            this.createSubscription();
            // re-trigger evaluation because observables use copies of arrays
            this.formControl.setValue(val);
          }
        }),
        takeUntil(this.destroy$)
      )
      .subscribe((newValue) => {
        // whatever
      });

    // goes along with tap
    // @ts-ignore
    this.subscription = this.formControl.valueChanges.observers[this.formControl.valueChanges.observers.length - 1];
  }
```

**Comment by aitboudad** [comment]
I'll check if there is a simple fix 😄 first.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.5.11](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.5.11) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
